src/storage/src/storage_state.rs

```diff
@@ -308,11 +310,20 @@ pub struct StorageState {
     /// Statistics for sources and sinks.
     pub aggregated_statistics: AggregatedStatistics,
 
-    /// Status updates reported by health operators.
+    /// A place shared with running dataflows, so that health operators, can
+    /// report status updates back to us.
     ///
     /// **NOTE**: Operators that append to this collection should take care to only add new
     /// status updates if the status of the ingestion/export in question has _changed_.
-    pub object_status_updates: Rc<RefCell<Vec<StatusUpdate>>>,
+    pub shared_status_updates: Rc<RefCell<Vec<StatusUpdate>>>,
+
+    /// The latest status update for each object.
+    pub latest_status_updates: BTreeMap<GlobalId, StatusUpdate>,
+
+    /// The latest status update that has been _reported_ back to the
+    /// controller. This will be reset when a new client connects, so that we
+    /// can determine what updates we have to report again.
+    pub reported_status_updates: BTreeMap<GlobalId, StatusUpdate>,
```

Comment:
The existence of this map surprises me! Naively I'd have thought that we can just unconditionally send the contents of `latest_status_updates` on reconnect. Or, if that's somehow hard to set up, wouldn't an `initial_status_reported` flag suffice?